### ISSUE TITLE
Add support for timeout configuration for OkHttpSender

### ISSUE THREAD

**Issue by pathikmehta** [openzipkin/zipkin-reporter-java#104]
Currently, we can configure `OkHttpSender` by configuring `OkHttpSenderFactoryBean` bean in XML configuraiton:
```xml
<bean id="spanReporter" class="zipkin2.reporter.beans.AsyncReporterFactoryBean">
    <property name="sender">
      <bean class="zipkin2.reporter.beans.OkHttpSenderFactoryBean">
       <property name="endpoint" value="http://localhost:9411/api/v2/spans"/>
     </bean>
   </property>
 </bean>
```

It has configurable parameters for `endpoint, maxRequests, compressionEnabled, messageMaxBytes` but it does not have `timeout` related parameters available to be configured through Spring-beans xml configurations.
Using Java configurations it might be possible but using xml configuration there seems no direct way to do so.

Similarly, for `URLConnectionSender` we can create bean for `URLConnectionSenderFactoryBean` which does provide the `timeout` related configuratble parameters like `connectTimeout, readTimeout`.

So it can be good if we have such `timeout` related configurable parameters available in spring-beans xml configurations. So that we can use `OkHttpSender` with `timeout` configurations using xml.

**Comment by adriancole** [comment]
looking at httpurlconnection, we are inconsistent with this. Will make it similar!

**Comment by adriancole** [comment]
accidentally fixed in this commit dcf0fec5b2ccf5dead5a99fa014761b8306692a4#diff-fe158f5084ddf2de348d4f8a7fec32f8R90 meant to make a different one.. whoops

**Comment by pathikmehta** [comment]
@adriancole Thank you for a quick fix.
